vuln_checker/main.py

```diff
@@ -172,6 +174,83 @@ def combine_vulnerabilities(vulns: VulnMap) -> VulnMap:
     return traverse_graph_and_combine_vulnerabilities(graph)
 
 
+def dump_vulns(vulns: VulnMap):
+    with (VULNS_PATH / "combined.json").open("w") as f:
+        json.dump(vulns, f, default=set_to_list, indent=2)
+
+
+Advisory = TypedDict(
+    'Advisory', {
+        'source': str,
+        'id': str,
+        'severity': str | None,
+        'cvss_v3_vector': str | None,
+        'cvss_v3_score': float | None
+    }
+)
+
+GroupedAdvisories = dict[str, list[Advisory]]
+DifferentAdvisories = TypedDict(
+    'DifferentAdvisories', {'severity': list[GroupedAdvisories]}
+)
+
+KeyForCheck = Literal['severity', 'cvss_v3_vector', 'cvss_v3_score']
+
+
+def compare_advisory_by_field_name(advisories: Iterable[Advisory], field_name: KeyForCheck) -> GroupedAdvisories:
+    key = itemgetter(field_name)
+    advisories = [advisory for advisory in advisories if advisory[field_name]]
+
+    sorted_advisories = sorted(advisories, key=key)
+    grouped = itertools.groupby(sorted_advisories, key)
+    return {k: list(g) for (k, g) in grouped}
+
+
+def advisory_to_short_info(advisory: Advisory) -> str:
+    source = advisory["source"].split("/")
+    identifier = source[-1][:-5]
+    database = source[-2]
+    return f"{identifier} in {database}"
+
+
+keys_for_check: tuple[Literal['severity']] = (
+    "severity",
+)
+
+
+def find_different_advisories(vulns: VulnMap) -> DifferentAdvisories:
+    different_advisories = {k: [] for k in keys_for_check}
+
+    for vuln in vulns.values():
+        # there is nothing to compare
+        if len(vuln["aliases"]) < 1:
+            continue
+
+        advisories: list[Advisory] = []
+        for source in vuln["sources"]:
+            with open(source, "r") as f:
+                advisories += [json.load(f) | {"source": source}]
+
+        for key in keys_for_check:
+            different = compare_advisory_by_field_name(advisories, key)
+            if len(different) > 1:
+                different_advisories[key].append(different)
+
+    return different_advisories
+
+
+def report_to_file(advisories: DifferentAdvisories):
```

Comment:
also i think we have to improve `find_different_advisories` and `report_to_file`.
so let's write the tests for these functions.